### PR TITLE
[Jonathan Tay] iP

### PR DESCRIPTION
**Pull request by jontmy**
### Changes

> v0.3.0

- Added a GUI to Duke wrapping around the CLI

### Tasks

- [ ] Refactor the UI code for proper OOP

### Miscellany

1. Here is a snippet of a [test](https://github.com/jontmy/ip/blob/master/src/test/java/duke/task/TodoTest.java) using `JUnit5`:
```java
public class TodoTest {
    @Test
    public void testToString() {
        assertEquals("[T][ ] abc", new Todo("abc").toString());
    }
}
```

**HAND** 👋

### CODE REVIEW

File: src/main/java/duke/command/UnmarkCommand.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * A command to unmark a task in the task list as done.
+ * Usage: unmark [0]
+ * [0]: index of the task to unmark as done
+ * Note that the index starts at 1.
+ */
+public class UnmarkCommand extends Command {
```
Comment (farisanadia) [inline src/main/java/duke/command/UnmarkCommand.java:16]: maybe this should be unmarkCommand

File: src/main/java/duke/internal/Parser.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
+package duke.internal;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
```
Comment (farisanadia) [inline src/main/java/duke/internal/Parser.java:5]: maybe you shouldn't leave a space here for consistency purposes?

File: src/main/java/duke/internal/Parser.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+ * and determine which Command object should be used to execute it.
+ */
+public class Parser {
+    /**
+     * Parses a command from a string.
+     *
+     * @param str the string to parse
+     * @return the Command object to execute
+     */
+    public static Command parse(String str) {
```
Comment (farisanadia) [inline src/main/java/duke/internal/Parser.java:28]: could this be called parseCommand instead? 
_"Names representing methods must be verbs and written in camelCase."_

File: src/main/java/duke/internal/Storage.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
```
Comment (farisanadia) [inline src/main/java/duke/internal/Storage.java:11]: similarly, maybe auto ordering can be used here for consistent import statements

File: src/main/java/duke/internal/Storage.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    /**
+     * Writes the list of tasks to a file at the given path.
+     * If the file already exists, it will be overwritten.
+     *
+     * @throws IOException if an I/O error occurs writing to the file
+     * @since Level-7
+     */
+    public void save(TaskList tasks) throws IOException {
```
Comment (farisanadia) [inline src/main/java/duke/internal/Storage.java:44]: could this be called saveTask and loadTask instead?

File: src/main/java/duke/task/Event.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+
+/**
+ * A task that start at a specific date/time and ends at a specific date/time.
+ */
+public class Event extends Task {
+    private final LocalDateTime datetime;
```
Comment (farisanadia) [inline src/main/java/duke/task/Event.java:9]: should this be called dateTime instead?
_" Variable names must be in camelCase."_

Comment (farisanadia) [inline src/main/java/duke/task/Event.java]: should this be called dateTime instead?
_" Variable names must be in camelCase."_

File: src/main/java/duke/internal/Storage.java (hunk 5 above)
Comment (farisanadia) [inline src/main/java/duke/internal/Storage.java]: could this be called saveTask and loadTask instead?

File: src/main/java/duke/internal/Storage.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
+package duke.internal;
+
+import java.io.IOException;
```
Comment (farisanadia) [inline src/main/java/duke/internal/Storage.java:3]: similarly, maybe auto ordering can be used here for consistent import statements

File: src/main/java/duke/internal/Parser.java (hunk 3 above)
Comment (farisanadia) [inline src/main/java/duke/internal/Parser.java]: could this be called parseCommand instead? 
_"Names representing methods must be verbs and written in camelCase."_

File: src/main/java/duke/internal/Parser.java [hunk 8]
```diff
@@ -0,0 +1,145 @@
+package duke.internal;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
```
Comment (farisanadia) [inline src/main/java/duke/internal/Parser.java]: maybe you shouldn't leave a space here for consistency purposes?

**Review by farisanadia — COMMENTED** [review]
Overall, your code is easy to understand with good javadocs. Well done!

File: src/main/java/duke/internal/Parser.java (hunk 8 above)
Comment (jontmy) [reply src/main/java/duke/internal/Parser.java]: Hi! Just wanted to point out that the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#package-and-import-statements) actually uses newlines to group related packages together.

The AB3 checkstyle also enforces this newline:
 
Screenshot 2022-09-03 at 13 13 05

Comment (farisanadia) [reply src/main/java/duke/internal/Parser.java]: ah i see! my bad

File: src/main/java/duke/internal/Storage.java (hunk 5 above)
Comment (jontmy) [reply src/main/java/duke/internal/Storage.java]: Good suggestion! I'll amend this in the next iteration.

**Pull request closed without merging**